### PR TITLE
Add description for the "columns" arguments

### LINKED ISSUE #764: Description for the "columns" arguments is missing
Please describe what could be improved about this page or the typo/mistake that you found:

I would suggest to add a short note regarding the supported syntax for the `columns` option when using e.g. a data file as input, something like 

* **data** (_str_ or _2d array_) – Either a data file name or a 2d numpy array with the tabular data. Use option _columns_ (i) to choose which columns are x, y, color, and size, respectively (e.g. `columns = [0, 1]` if the x values are stored in the first column and y values in the second one, note: zero-based indexing is used).

I tried to fix that suggestion by my own following the instructions given in the **Contributing Guidelines** under _Editing the Documentation_, however, the corresponding `.rst` file is not available in the `doc` folder.

### PR DESCRIPTION
**Pull request by michaelgrund**
Add missing description for the columns argument in `Figure.plot`.

Closes #764.

### CODE REVIEW

File: pygmt/base_plotting.py [hunk 1]
```diff
@@ -782,6 +782,11 @@ def plot(self, x=None, y=None, data=None, sizes=None, direction=None, **kwargs):
             polygon in the input data. To apply it to the fill color, use
             ``color='+z'``. To apply it to the pen color, append **+z** to
             **pen**.
+        columns : str or 1d array
+            Choose which columns are x, y, color, and size, respectively if input
+            is provided via *data*. E.g. ``columns = [0, 1]`` or ``columns = '0, 1'``
```
Comment (seisman) [inline pygmt/base_plotting.py:787]:
```suggestion
            is provided via *data*. E.g. ``columns = [0, 1]`` or ``columns = '0,1'``
```

File: pygmt/base_plotting.py [hunk 2]
```diff
@@ -782,6 +782,11 @@ def plot(self, x=None, y=None, data=None, sizes=None, direction=None, **kwargs):
...
+            if the *x* values are stored in the first column and *y* values in the
+            second one: Note: zero-based indexing is used.
```
Comment (seisman) [inline pygmt/base_plotting.py:789]:
```suggestion
            second one. Note: zero-based indexing is used.
```

**Comment by seisman** [comment]
@michaelgrund Thanks for your PR.

Your `/format` command doesn't work due to a bug (https://github.com/GenericMappingTools/pygmt/issues/767), and I will try to fix it later.

I just made two tiny comments about your changes. Besides that, you also need to manually limit the docstrings to no more than 80 characters per line to fix these styling issues:
```
flake8 pygmt setup.py doc/conf.py
pygmt/base_plotting.py:786:80: W505 doc line too long (81 > 79 characters)
pygmt/base_plotting.py:787:80: W505 doc line too long (85 > 79 characters)
pygmt/base_plotting.py:788:80: W505 doc line too long (82 > 79 characters)
make: *** [check] Error 1
Makefile:44: recipe for target 'check' failed
```

File: pygmt/base_plotting.py (hunk 1 above)
Comment (michaelgrund) [reply pygmt/base_plotting.py:787]: Fine from my side.

File: pygmt/base_plotting.py (hunk 2 above)
Comment (michaelgrund) [reply pygmt/base_plotting.py:789]: Fine from my side.

**Comment by michaelgrund** [comment]
> 
> 
> @michaelgrund Thanks for your PR.
> 
> Your `/format` command doesn't work due to a bug (#767), and I will try to fix it later.
> 
> I just made two tiny comments about your changes. Besides that, you also need to manually limit the docstrings to no more than 80 characters per line to fix these styling issues:
> 
> ```
> flake8 pygmt setup.py doc/conf.py
> pygmt/base_plotting.py:786:80: W505 doc line too long (81 > 79 characters)
> pygmt/base_plotting.py:787:80: W505 doc line too long (85 > 79 characters)
> pygmt/base_plotting.py:788:80: W505 doc line too long (82 > 79 characters)
> make: *** [check] Error 1
> Makefile:44: recipe for target 'check' failed
> ```

Thanks for fixing it. Totally read over docstrings limit, good to know for next time ;)

File: pygmt/base_plotting.py [hunk 3]
```diff
@@ -782,6 +782,11 @@ def plot(self, x=None, y=None, data=None, sizes=None, direction=None, **kwargs):
             polygon in the input data. To apply it to the fill color, use
             ``color='+z'``. To apply it to the pen color, append **+z** to
             **pen**.
+        columns : str or 1d array
+            Choose which columns are x, y, color, and size, respectively if input
+            is provided via *data*. E.g. ``columns = [0, 1]`` or ``columns = '0,1'``
+            if the *x* values are stored in the first column and *y* values in the
+            second one. Note: zero-based indexing is used.
```
Comment (seisman) [inline pygmt/base_plotting.py:789]:
```suggestion
            Choose which columns are x, y, color, and size, respectively if 
            input is provided via *data*. E.g. ``columns = [0, 1]`` or 
            ``columns = '0,1'`` if the *x* values are stored in the first 
            column and *y* values in the second one. Note: zero-based 
            indexing is used.
```

**Pull request merged**